templates/deno/src/services/service.ts.twig

```diff
@@ -69,8 +70,83 @@ export class {{ service.name | caseUcfirst }} extends Service {
         if (typeof {{ parameter.name | caseCamel | escapeKeyword }} !== 'undefined') {
             payload['{{ parameter.name }}'] = {{ parameter.name | caseCamel | escapeKeyword }};
         }
+{% endfor %}
+{% if 'multipart/form-data' in method.consumes %}
+{% for parameter in method.parameters.all %}
+{% if parameter.type == 'file' %}
+        const {size: size} = await Deno.stat({{ parameter.name | caseCamel | escapeKeyword }});
 
+        if (size <= Client.CHUNK_SIZE) {
+            payload['{{ parameter.name }}'] = new File([await Deno.readFile({{ parameter.name | caseCamel | escapeKeyword }})], basename({{ parameter.name | caseCamel | escapeKeyword }}));
+
+            return await this.client.call('{{ method.method | caseLower }}', path, {
+{% for parameter in method.parameters.header %}
+                '{{ parameter.name }}': ${{ parameter.name | caseCamel | escapeKeyword }},
+{% endfor %}
+{% for key, header in method.headers %}
+                '{{ key }}': '{{ header }}',
+{% endfor %}
+            }, payload);
+        } else {
+            const stream = await Deno.open({{ parameter.name | caseCamel | escapeKeyword }}, { read: true });
+
+            let id = undefined;
+            let response = undefined;
+
+            const totalCounters = Math.ceil(size / Client.CHUNK_SIZE);
+
+            for (let counter = 0; counter < totalCounters; counter++) {
+                    const start = (counter * Client.CHUNK_SIZE);
+                    const end = Math.min((((counter * Client.CHUNK_SIZE) + Client.CHUNK_SIZE) - 1), size);
+                    const headers: { [header: string]: string } = {
+{% for parameter in method.parameters.header %}
+                            '{{ parameter.name }}': ${{ parameter.name | caseCamel | escapeKeyword }},
+{% endfor %}
+{% for key, header in method.headers %}
+                            '{{ key }}': '{{ header }}',
 {% endfor %}
+                            'content-range': 'bytes ' + start + '-' + end + '/' + size
+                    };
+
+                    if (id) {
+                            headers['x-appwrite-id'] = id;
+                    }
+                    
+                    const totalBuffer = new Uint8Array(Client.CHUNK_SIZE);
+
+                    // 16kB chunks block limit: https://github.com/denoland/deno/discussions/9906
```

Comment:
Can we move the comment to where we set `Client.CHUNK_SIZE`

Reply:
Ah yes, it's there already, we need to remove this @Meldiron

Reply:
Removed